File: ToDoList/ToDoList.DAL/Models/TaskItem.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using ToDoList.DAL.Models.Enums;

namespace ToDoList.DAL.Models
{
    public class TaskItem
    {
        public int TaskItemId { get; set; }

        public string Name { get; set; }

        public TaskItemStatus Status { get; set; }

        public string? Description { get; set; }
    }
}

File: client-app/src/app/models/TaskItem.ts
export interface TaskItem {
    taskItemId: number;
    name: string;
    description?: string;
    status: number;
}

export class TaskItem implements TaskItem {
    constructor() {
        this.taskItemId = 0;
        this.name = "";
        this.status = 0;
        this.description = undefined;
    }
}